pkg/web/apps/app_service_moq_test.go

```diff
@@ -0,0 +1,70 @@
+// Code generated by moq; DO NOT EDIT.
```

Comment:
What benefit are we getting out of this additional file?

Reply:
The file has the mock implementation. it is called/used in the tests. You can check if we remove this file then the test will fail because no longer the implementation is in the project. 

```
$ make test
Running tests:
GOCACHE=off go test -cover \
	  github.com/aerogear/mobile-security-service/pkg/config github.com/aerogear/mobile-security-service/pkg/db github.com/aerogear/mobile-security-service/pkg/httperrors github.com/aerogear/mobile-security-service/pkg/test github.com/aerogear/mobile-security-service/pkg/web/apps
ok  	github.com/aerogear/mobile-security-service/pkg/config	0.043s	coverage: 100.0% of statements
?   	github.com/aerogear/mobile-security-service/pkg/db	[no test files]
# github.com/aerogear/mobile-security-service/pkg/web/apps [github.com/aerogear/mobile-security-service/pkg/web/apps.test]
pkg/web/apps/apps_service_test.go:22:23: undefined: AppServiceMock
ok  	github.com/aerogear/mobile-security-service/pkg/httperrors	0.056s	coverage: 70.4% of statements
?   	github.com/aerogear/mobile-security-service/pkg/test	[no test files]
FAIL	github.com/aerogear/mobile-security-service/pkg/web/apps [build failed]
make: *** [test-unit] Error 2
```

<img width="595" alt="screenshot 2019-02-19 at 12 01 46" src="https://user-images.githubusercontent.com/7708031/53013654-3565ee00-343e-11e9-814c-faab46150534.png">

Is it make sense?

Reply:
Yes it makes sense. I wonder if it can be all combined into the same test file. I will be tackling this for my own endpoint in the next day or so and will learn more about it.

Reply:
What single file would they be in @davidffrench?

Reply:
I would rather if it was possible to have all unit tests for a single file contained in a single unit test file. This does not seem to be how the mocks work, I need to look into them more.

Reply:
The mock file doesn't have any unit tests though - it is different. It mocks the service so that the tests can interact with them as if they were the real service. It would be misleading having it in the test file.

See how Integreatly use it: [realm/handler_moq.go](https://github.com/integr8ly/keycloak-operator/blob/bb03fa20279a3f88159f469f308c0d1da8c62a05/pkg/keycloak/realm/handler_moq.go)

Reply:
Thank you @craicoverflow

Reply:
@davidffrench if we combine the file into the test then it will lose the advantage to use the lib/dep. The lib dep will create/update the mock file automatically to be used in the tests. Could we agree in move forward in this way, at least for now?

Reply:
Absolutely. Please proceed

Reply:
Thank you 👍